### PR TITLE
[eventsource] compatibility with non-DOM consumers

### PR DESCRIPTION
**Pull request by Akuukis**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [You can use [eventsource] with Node.js or as a browser polyfill](https://github.com/EventSource/eventsource)

### CODE REVIEW

**Comment by scottleedavis** [comment]
Does this remove the need for the majority of the polyfill @akuukis?

**Comment by Akuukis** [comment]
@scottleedavis No, but as `eventsource-polyfill/index.d.ts` was identical to `index.d.ts` I just re-exported it to keep code DRY. It's an unrelated refactor to "DOM-consumer compatibility" scope.

**Review by scottleedavis — APPROVED** [review]
lgtm.  👍

File: types/eventsource/index.d.ts [hunk 1]
```diff
@@ -3,8 +3,15 @@
 // Definitions by: Scott Lee Davis <https://github.com/scottleedavis>
 //                 Ali Afroozeh <https://github.com/afroozeh>
 //                 Pedro Gámez <https://github.com/snakedrak>
+//                 Akuukis <https://github.com/Akuukis>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.0
+
+// eventsource uses DOM dependencies which are absent in browserless envoronment like node.js.
```
Comment (SnakeDrak) [inline types/eventsource/index.d.ts]: Just a misspelled in the comment:
> [...] in browserless **envoronment** like [..]

**Review by SnakeDrak — APPROVED** [review]
:+1:

File: types/eventsource/index.d.ts [hunk 2]
```diff
@@ -3,8 +3,15 @@
 // Definitions by: Scott Lee Davis <https://github.com/scottleedavis>
 //                 Ali Afroozeh <https://github.com/afroozeh>
 //                 Pedro Gámez <https://github.com/snakedrak>
+//                 Akuukis <https://github.com/Akuukis>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.0
```
Comment (uniqueiniquity) [inline types/eventsource/index.d.ts]: Is there a particular reason the minimum required version needed to be updated? I'm not sure I see it.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/eventsource/index.d.ts (hunk 2 above)
Comment (Akuukis) [reply types/eventsource/index.d.ts]: @uniqueiniquity Yes, `/// <reference lib="..." />` comes with [TS 3.0](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html). It is used in DOM test file to check whenever monkeypatch is mergable with the full DOM typings.

There is nothing that impacts types themselves.

**Comment by Akuukis** [comment]
Bump! LGTM to merge, do I need to do anything?

P.S. @uniqueiniquity I answered your request and it doesn't need new commits.

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**